spock-core/src/main/java/org/spockframework/runtime/extension/builtin/UnrollExtension.java

```diff
@@ -14,39 +14,86 @@
 
 package org.spockframework.runtime.extension.builtin;
 
+import org.spockframework.runtime.DataVariablesIterationNameProvider;
+import org.spockframework.runtime.GroovyRuntimeUtil;
+import org.spockframework.runtime.InvalidSpecException;
 import org.spockframework.runtime.extension.AbstractAnnotationDrivenExtension;
+import org.spockframework.runtime.extension.AbstractGlobalExtension;
 import org.spockframework.runtime.model.FeatureInfo;
 import org.spockframework.runtime.model.IterationInfo;
 import org.spockframework.runtime.model.NameProvider;
 import org.spockframework.runtime.model.SpecInfo;
 
+import spock.lang.Rollup;
 import spock.lang.Unroll;
 
 public class UnrollExtension extends AbstractAnnotationDrivenExtension<Unroll> {
+  private final String globalUnrollPattern;
+
+  public UnrollExtension() {
+    String globalUnrollPattern = System.getProperty("spock.globalUnrollPattern");
```

Comment:
Spock has extension config objects for that

Reply:
Ah, nice, forgot about these.
I would suggest to pack `spock.globalUnrollPattern`, `spock.assertUnrollExpressions` and `spock.globalRollup` all in one extension that can then be used by both extensions (the third one actually).
How do we want to name it?
`UnrollConfiguration` with these options and defaults:
```groovy
unroll {
    globalPattern = null
    assertExpressions = false
    enable = true
}
```
?

Reply:
Collecting together sounds good for me.

I wonder if we could change `assertExpressions` to `true` by default. For every new project I don't see a good excuse to do not use it (therefore, enabled by default). For already existing projects it might be beneficial to get know about typos and others issues. For large problem with multiple errors, well, it can be disabled anyway with per project configuration (probably the exception should have a link to the documentation explaining how to do it). WDYT?

Reply:
If so, then now with 2.0 as it is a potentially breaking change.
I wouldn't object to do so, I just assumed it has a reason it is not on by default.
Maybe to not have a breaking change, but 1.2 which introduced the flag also had other breaking changes.

Reply:
If not in 2.0 then never. Changing the major version number is a good opportunity to enforce some good practices (which could be easily disabled for "legacy" projects).

Reply:
The suggested example looks fine and yes you can share the same object. See `IncludeExcludeExtension` and `OptimizeRunOrderExtension`

Reply:
Yeah, I know, that was not the question, just about the naming. :-)
What's your opinion about the changed `assertExpressions` default @leonard84?

Reply:
Hm, I don't like `enable` or `enabled`, it might give the impression, that you can disable unrolling alltogether, independent of annotations. Any suggestions for a better name for that option?

Reply:
currently I have
```groovy
unroll {
  enabled false
  defaultPattern '#featureName[#iterationIndex]'
  assertExpressions true
}
```

Reply:
How about `unroll { global false }`?

Reply:
Or `unroll { default false }` or `unroll { defaultUnroll false }`

Reply:
`unroll { enabledByDefault true }`?